src/openforms/submissions/api/views.py

```diff
@@ -104,5 +104,10 @@ def get(self, request, *args, **kwargs):
         )
 
     def perform_destroy(self, instance):
+        # delete files from disc as well if they had been already
+        # saved when trying to access the next form step
+        if instance.attachments.exists():
+            instance.attachments.all().delete()
```

Comment:
I think you can avoid a query by removing the `.exists`, or does that crash?

Reply:
I have understood that without calling any method the value will be always truthy. Tested locally and indeed it enters the if statement even if there are no attachments.

Reply:
Discussed with Sergei, the if statement here can be removed.